### PR TITLE
Address few misc. bugs

### PR DESCRIPTION
**Pull request by chesterxgchen**
Fixes # .
* Step-by-step example train_with_mlflow.py should use global_step for logging metrics
* fl_model_utils.update_model() losing metrics after update
 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by IsaacYangSLA — APPROVED** [review]
Thanks.  LGTM.

File: nvflare/app_common/utils/fl_model_utils.py [hunk 1]
```diff
@@ -55,8 +56,10 @@ def to_shareable(fl_model: FLModel) -> Shareable:
             raise ValueError("FLModel without params and metrics is NOT supported.")
         elif fl_model.params is not None:
             if fl_model.params_type is None:
-                raise ValueError(f"Invalid ParamsType: ({fl_model.params_type}).")
-            data_kind = params_type_to_data_kind.get(fl_model.params_type)
+                # raise ValueError(f"Invalid ParamsType: ({fl_model.params_type}).")
```
Comment (holgerroth) [inline nvflare/app_common/utils/fl_model_utils.py:59]: why keep this comment?

File: nvflare/app_common/utils/fl_model_utils.py [hunk 2]
```diff
@@ -103,15 +106,20 @@ def from_shareable(shareable: Shareable, fl_ctx: Optional[FLContext] = None) ->
...
+                        params_type = ParamsType.FULL
+
                 params_type = ParamsType(params_type)
 
-                params = dxo.data
                 if MetaKey.INITIAL_METRICS in meta:
                     metrics = meta[MetaKey.INITIAL_METRICS]
         except:
             # this only happens in cross-site eval right now
+            # todo: need to fix this. swallow exception to get here, seems really weired.
```
Comment (holgerroth) [inline nvflare/app_common/utils/fl_model_utils.py:122]: If we keep the todo, it should be "weird".

File: nvflare/app_common/utils/fl_model_utils.py [hunk 3]
```diff
@@ -205,6 +213,9 @@ def update_model(model: FLModel, model_update: FLModel, replace_meta: bool = Tru
             model.meta = model_update.meta
```
Comment (holgerroth) [inline nvflare/app_common/utils/fl_model_utils.py:213]: There's a typo in the RuntimeError above. The message should be `f"params_type {model.params_type} of `model`...`

**Review by holgerroth — CHANGES_REQUESTED** [review]
Added some comments with request for change.

**Review by holgerroth — APPROVED** [review]
Makes sense.

**Review by holgerroth — APPROVED** [review]

**Pull request merged**